src/ATen/native/xpu/Histogram.cpp

```diff
@@ -250,10 +267,165 @@ static Tensor& histogramdd_out(
     bin_edges[dim].copy_(bins[dim]);
   }
 
-  at::native::xpu::histogramdd_kernel(self, weight, density, hist, bin_edges);
+  const int64_t N = self.size(-1);
+  const int64_t M = std::accumulate(
```

Comment:
We'd better make frontend (non-kernel) impl the same as CPU. Because we have plan to register xxx_stub only. These non-kernel code will reuse CPU implementation.